src/main/java/mate/academy/bookshop/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package mate.academy.bookshop.service.impl;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.order.OrderItemResponseDto;
+import mate.academy.bookshop.dto.order.OrderRequestDto;
+import mate.academy.bookshop.dto.order.OrderResponseDto;
+import mate.academy.bookshop.dto.status.StatusDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.OrderItemMapper;
+import mate.academy.bookshop.mapper.OrderMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.Order;
+import mate.academy.bookshop.model.OrderItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.model.Status;
+import mate.academy.bookshop.repository.OrderRepository;
+import mate.academy.bookshop.repository.OrderStatusRepository;
+import mate.academy.bookshop.service.OrderService;
+import mate.academy.bookshop.service.ShoppingCartService;
+import mate.academy.bookshop.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private static final Status.StatusName DEFAULT_STATUS_NAME
+            = Status.StatusName.PENDING;

```

Comment:
I do not see a reason to make enum a constant, they are basically already a constants by their nature